usecase/patientData.go

```diff
@@ -232,6 +232,8 @@ type GetDataArgs struct {
 }
 
 func (p *PatientData) GetData(ctx context.Context, args GetDataArgs) (*bytes.Buffer, *common.DetailedError) {
+	common.TimeIt(ctx, "getData")
+	defer common.TimeEnd(ctx, "getData")
```

Comment:
with defer it will include time passed in writeDataToBuffer function call. 

Maybe we don't need it. The `channelReadLoop` timer should give us the time spent while reading data. `FillLoopModeEvents` & `CleanUpBasals` the time spent rebuilding basals. and `writeDataToBuffer` the total write time

Reply:
i hesitate but i find it useful to understand the timing in the method above (in the middleware function),  i prefer to keep it  as it avoid mental calculation.